dist/tools/ci/can_fast_ci_run.py

```diff
@@ -164,14 +164,39 @@ def classify_changes(riotbase=None, upstream_branch="master"):
             match = REGEX_GIT_DIFF_SINGLE_FILE.match(line)
             if match:
                 file = match.group(1)
-                change_set.add_file(file)
+
+                if only_comment_change(file, upstream_branch) is False:
+                    change_set.add_file(file)
+
                 continue
 
             raise Exception("Failed to parse \"{}\"".format(line))
 
     return change_set
 
 
+def get_hash_without_comments(file, ref):
+    result = subprocess.run(
+        f"git show {ref}:{file} | gcc -fpreprocessed -dD -E -P - | md5sum",
+        shell=True,
+        capture_output=True,
+        check=True,
+    )
+    return result.stdout
+
+
+def only_comment_change(file, upstream_branch):
+    try:
+        if file[-2:] in {".c", ".h"}:
```

Comment:
Using `os.path.splitext`, you could check for more file extensions: `hpp`, `cpp`. Maybe use a constant for the files extensions:

```python
FILE_EXTENSIONS_FOR_COMMENTS = [".c", ".h", ".hpp", ".cpp"]
[....]

def only_comment_change(file, upstream_branch):
[...]
    if os.path.splitext(file)[1] in FILE_EXTENSIONS_FOR_COMMENTS:
        [...]
```